src/react-native/11-navigation/navigation.md

```diff
@@ -64,11 +161,73 @@ If you’ve implemented the solution correctly, the tests will pass when you run
 
 ## Objective 2: Using React Native stack navigation
 
-TODO
-
-### Concept TODO
-
-TODO
+Now that we've covered tab navigation, let's move on to stack navigation. As mentioned in the previous section, most applications use a combination of both tab and stack navigation to create a seamless user experience.
+
+### Stack vs Native Stack
+
+The React Navigation library provides both a stack navigator as well as a native stack navigator. The native stack navigator uses the native APIs provided by the platform, `UINavigationController` on iOS and `Fragment` on Android. Because it is using the native APIs, it provides native performance and exposes native-specific features. Although the stack navigator is not as performant, it is an easier-to-use, more customizable, JavaScript-based alternative.
```

Comment:
This description makes me think we should’ve used the native stack. 😅 I wonder if we should sell the JS stack harder:

- Flexibility and Customization: The JS navigator offers more flexibility in customization. You can tweak animations, transitions, and behaviors directly in JavaScript, which can be advantageous if you need highly customized navigation flows that aren't easily achieved with native components.

- Simpler Setup and Debugging: Since JS navigators are implemented entirely in JavaScript, you don’t have to deal with native code, making setup, maintenance, and debugging generally simpler. This can speed up development, especially if your team specializes in JavaScript and does not have as much experience with native mobile development.

- Consistency Across Platforms: JS-based navigators can offer a more consistent look and feel across different platforms (Android and iOS), as the same JavaScript code controls the navigation on both platforms. This can be important for maintaining brand consistency or when you want to ensure the user experience is the same, regardless of the device.

Reply:
love this writeup 😄 updated